parts/router.go

```diff
@@ -1906,6 +1913,20 @@ func (router *Router) prepareMcRequest(colIds []uint32, firstReq *base.WrappedMC
 			reqToProcess.ColInfo.TargetNamespace = migrationNamespaceMap[colId]
 		}
 		reqToProcess.ColInfoMtx.Unlock()
+
+		if targetManifest == nil {
+			continue
```

Comment:
> It could be dangerous that future code added will not execute due to this continue
Better to run the code 1921 and beyond with targetManifest != nil
Done - changed to:
```
if targetManifest != nil {
	scopeName, collectionName, err := targetManifest.GetScopeAndCollectionName(colId)
	if err == nil {
		reqToProcess.TgtColNamespaceMtx.Lock()
		reqToProcess.TgtColNamespace = &base.CollectionNamespace{
			ScopeName:      scopeName,
			CollectionName: collectionName,
		}
		reqToProcess.TgtColNamespaceMtx.Unlock()
	}
}
```
> However, it may be safer to always pass a non-nil target manifest.
The only time I can think of targetManifest to be a nil is if the source namespace has already been marked denied, which means it wasn't even in the rules to be matched.

> For implicit mapping, just pass a defaultManifest... then you won't need to check for a non-nil target manifest
I don't really get this point. How will target manifest for implicit mapping be default manifest?

Reply:
targetManifest != nil should be present I think irrespective of mapping mode etc, atleast for the sake of safety.